main.sh

```diff
@@ -36,15 +39,35 @@ if [[ $# -gt 0 ]]; then
     bail "invalid argument '$1'"
 fi
 
+dry_run="${INPUT_DRY_RUN:-}"
+case "${dry_run}" in
+    true) dry_run="1" ;;
+    false) dry_run="" ;;
+    *) bail "'dry_run' input option must be 'true' or 'false': '${dry_run}'" ;;
+esac
+
 token="${INPUT_TOKEN:-"${GITHUB_TOKEN:-}"}"
 ref="${INPUT_REF:-"${GITHUB_REF:-}"}"
 
 if [[ -z "${token}" ]]; then
-    bail "neither GITHUB_TOKEN environment variable nor 'token' input option is set"
+    if [[ -n "${dry_run}" ]]; then
+        # TODO: Downgrade to info instead of warning because of more noise on
+        # https://github.com/<repo>/actions/runs/<id> page when set to warning?
```

Comment:
Currently, the warnings are somewhat noisy if we have a lot of build matrix.
https://github.com/taiki-e/upload-rust-binary-action/actions/runs/6452798607

However, given the risk of making dry-run useless by missing a problem if there is no warning, we did not default to info (which is output in the log but is not very noticeable).

Perhaps adding another option to downgrade the warning might make sense.